### DISCUSSION TITLE
Demo functions

### DISCUSSION THREAD

**Discussion by Agrover112** [guofei9987/scikit-opt#137] [Ideas]
I had a few ideas regarding functions to add to the repository.
I'm not sure how this would break the CI/CD as it did in my commit which was just a fomatting.
- But I was thinking we should re-arrange the functions in the file according to their Shape?
- Also I encountered few function variants of the ones which are already implemented with noise?
If required one could add them as well?

**Comment by Agrover112** [comment]
@guofei9987

**Reply by guofei9987** [reply]
Firstly thank you for your merge request!

It break the CI/CD because the code coverage rate decreased. 
Just ignore the warning if you know what is going on.

New functions can also be added here https://github.com/guofei9987/scikit-opt/blob/master/examples/obj_func_demo.py

**Reply by Agrover112** [reply]
Why should we add in obj_func_demo? when another file called demo_func already exists? Any logical reasoning behind having two files?
I would suggest having a single file with all fns. Otherwise people using the api might get confused or find a missing import while importing from demo_func and obj_demo_func_demo .

**Reply by guofei9987** [reply]
When do `pip install scikit-opt`, all files in sko/ will be installed. So, There is no need to make the `demo_func.py` too big.

**Reply by Agrover112** [reply]
Ok no problem. I will add other if I find any in the examples. 
Do you mind if I add more details to demo_func  related to the func shape, type in docstring?
Also I need to add refs for the function in the Readme.
